01_fib/jimbotech/ex-fibonacci.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// fibonacci returns a function that returns
+// an number in the fibonacci sequence.
+// This is an implmentation of "closure".
+// The inside function returns the current number
+// in the fibonnacci sequence, starting at 1
+// and sets up the next which will be returned
+// in the subsequent call
+func fibonacci() func() int {
+	secondLast, last := 0, 1
+
+	return func() int {
+		result := last
+		secondLast, last = last, secondLast+last
+		return result
+	}
+}
+
+func fib(n int) {
+	f := fibonacci()
+
+	if n < 0 {
+		for i := 0; i > n; i-- {
```

Comment:
There is more idiomatic way of writing [for loops ](https://golang.org/doc/effective_go.html) and you have used already on line 14 of the test file.

Reply:
In line 14 of my test I was "ranging" over a collection. This is not the case here. I am simply counting from 0 to n. I can't think of any other way to do it, unless you could show me how?